x-pack/plugin/vectors/src/main/java/org/elasticsearch/xpack/vectors/VectorsUsageTransportAction.java

```diff
@@ -41,8 +47,42 @@ public VectorsUsageTransportAction(TransportService transportService, ClusterSer
     @Override
     protected void masterOperation(Task task, XPackUsageRequest request, ClusterState state,
                                    ActionListener<XPackUsageFeatureResponse> listener) {
+        boolean vectorsAvailable = licenseState.isVectorsAllowed();
+        boolean vectorsEnabled = XPackSettings.VECTORS_ENABLED.get(settings);
+        int numDenseVectorFields = 0;
+        int numSparseVectorFields = 0;
+        int avgDenseVectorDims = 0;
+
+        if (vectorsAvailable && vectorsEnabled) {
+            for (IndexMetaData indexMetaData : state.metaData()) {
```

Comment:
this looks like an expensive operation to process all mappings of all indexes to search for `dense_vector` and `sparse_vector` types. But I don't know how to gather statistics otherwise.